pkgs/applications/misc/blender/default.nix

```diff
@@ -1,243 +1,54 @@
-{ config, stdenv, lib, fetchurl, fetchzip, boost, cmake, ffmpeg, gettext, glew
-, ilmbase, libepoxy, libXi, libX11, libXext, libXrender
-, libjpeg, libpng, libsamplerate, libsndfile
-, libtiff, libwebp, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio, openjpeg, python310Packages
-, openvdb, libXxf86vm, tbb, alembic
-, zlib, zstd, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
-, jackaudioSupport ? false, libjack2
-, cudaSupport ? config.cudaSupport, cudaPackages ? { }
-, hipSupport ? false, rocmPackages # comes with a significantly larger closure size
-, colladaSupport ? true, opencollada
-, spaceNavSupport ? stdenv.isLinux, libspnav
-, makeWrapper
-, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
-, waylandSupport ? stdenv.isLinux, pkg-config, wayland, wayland-protocols, libffi, libdecor, libxkbcommon, dbus
-, potrace
-, openxr-loader
-, embree, gmp, libharu
-, openpgl
-, mesa
-, runCommand
+{ config, lib, stdenv, llvmPackages_10, darwin, callPackage, python310Packages
+, jackaudioSupport ? false
+, cudaSupport ? config.cudaSupport
+, hipSupport ? false  # comes with a significantly larger closure size
+, colladaSupport ? true
+, spaceNavSupport ? stdenv.isLinux
+, waylandSupport ? stdenv.isLinux
 }:
 
 let
-  python = python310Packages.python;
-  optix = fetchzip {
-    # url taken from the archlinux blender PKGBUILD
-    url = "https://developer.download.nvidia.com/redist/optix/v7.3/OptiX-7.3.0-Include.zip";
-    sha256 = "0max1j4822mchj0xpz9lqzh91zkmvsn4py0r174cvqfz8z8ykjk8";
-  };
-
+  versions = lib.importJSON ./versions.json;
+
+  latestStable = lib.last (builtins.sort lib.versionOlder (builtins.attrNames versions));
+  latestLTS = lib.last (builtins.sort lib.versionOlder (builtins.attrNames (lib.filterAttrs (_: v: v.isLTS == true) versions)));
+
+  packages =
+    lib.mapAttrs'
+    (version: content: {
+      name = "blender_${version}";
+      value = callPackage ./derivation.nix {
+        inherit (darwin.apple_sdk.frameworks) Cocoa CoreGraphics ForceFeedback OpenAL OpenGL;
+
+        # LLVM 11 crashes when compiling GHOST_SystemCocoa.mm
+        stdenv = if stdenv.isDarwin then llvmPackages_10.stdenv else stdenv;
+
+        # TODO Consider following https://vfxplatform.com/ for dependency tracking
+        # Like "VFX_RefCY": "2024", which outlines python 3.11, qt 6.5+ etc
+        pythonPackages = python310Packages;
+
+        withPackages = callPackage ./wrapper.nix {
+          blender = packages."blender_${version}";
+          pythonPackages = packages."blender_${version}".pythonPackages;
+        };
+
+        inherit jackaudioSupport hipSupport cudaSupport colladaSupport spaceNavSupport waylandSupport;
```

Comment:
You don't have to manually pass `cudaSupport`&c into `callPackage` arguments, you can just move all of the `config` and `isLinux` refernces to your `generic.nix`